src/modules/water-demand-prediction/water-demand-prediction.component.ts

```diff
@@ -0,0 +1,545 @@
+          text: "Time",
+        },
+        grid: {
+          display: false, // Show grid lines on the y-axis
+          color: "#000000", // Customize the grid line color #000000
+          lineWidth: 0.4, // Set the width of the grid lines
+        },
+      },
+    },
+  };
+
+  /**
+   * The chart object, referenced from the html template.
+   * ViewChildren is a list of charts, because when using ViewChild,
+   * only ever the first chart gets updated at all.
+   */
+  @ViewChildren(BaseChartDirective) charts:
+    | QueryList<BaseChartDirective>
+    | undefined;
+
+  constructor(private translateService: TranslateService) {
+    dayjs.locale(this.translateService.currentLang || "en");
+
+    // update dayjs whenever language changes
+    this.translateService.onLangChange.subscribe(event => {
+      dayjs.locale(event.lang);
+    });
+
+    /** define smartmeter options */
+    effect(() => {
+      let smartmeters = this.smartmeterSignal();
+      if (smartmeters) {
+        this.optionsSmartmeter = smartmeters;
+      }
+    });
+
+    /** updates optionsWeatherColumn when selected weather attribute changes */
+    effect(() => {
+      let weatherCols = this.weatherColumnsSignal();
+      if (weatherCols) {
+        this.optionsWeatherColumn = weatherCols;
+      }
+    });
+
+    /** request current smartmeter data */
+    effect(() => {
+      let curData = this.currentSingleSmartmeterDataSignal();
+      if (curData) {
+        this.currentSmartmeterData = curData;
+      }
+    });
+
+    /** request predicted smartmeter data */
+    effect(() => {
+      if (this.triggerFetchPredictedSmartmeterData()) {
+        let curData = this.currentPredictedSmartmeterDataSignal();
+        this.currentPredictedSmartmeterData = curData;
+      }
+    });
+
+    effect(() => {
+      let info = this.trainingResp();
+      if (info) {
+        this.modalOpen.set(true);
+      }
+    });
+  }
+
+  /** set displayed resolution and update chart to mirror that */
+  protected setDisplayedResolution(resolution: string): void {
+    this.displayedResolution.set(resolution);
+  }
+
+  /**
+   * update all charts in ViewChildren
+   * update only one chart based on index given, when given one.
+   * @returns if there is no chart
+   */
+  protected updateCharts(indexOfChart?: number): void {
+    if (!this.charts) {
+      console.log("No chart initialized!");
+      return;
+    }
+
+    // add 1 to update the first graph, add 2 to update the prediction graph only
+    if (indexOfChart) {
+      this.charts.toArray()[indexOfChart].update();
+      return;
+    }
+
+    this.charts.forEach(child => {
+      if (!child.chart) {
+        console.log("No chart in charts!");
+        return;
+      }
+      child.chart.update();
+    });
+  }
+
+  /** Completely erases data from the real data graph element */
+  protected resetChart(): void {
+    this.chartDataCurrentValues.datasets = [];
+
+    this.savedDatasets = {};
+    this.updateCharts();
+  }
```

Comment:
This sounds like incorrect use of the charts, manually updating charts is rarely needed